src/components/StoreMigrator/index.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useState } from 'react'
+import { useDispatch } from 'react-redux'
+import { addressBookImport } from 'src/logic/addressBook/store/actions'
+import { logError, Errors } from 'src/logic/exceptions/CodedException'
+import { MigrationMessage } from 'src/routes/migration/container'
+import { MIGRATION_ADDRESS } from 'src/routes/routes'
+
+const MAINET_URL = 'http://localhost:3000'
+const networks = [
+  {
+    safeUrl: 'http://localhost:3001/#',
+  },
+  {
+    safeUrl: 'http://localhost:3002/#',
+  },
+]
+
+type MigrationMessageEvent = MessageEvent & {
+  data: MigrationMessage
+}
+
+const StoreMigrator: React.FC = () => {
+  const [currentNetwork, setCurrentNetwork] = useState(0)
+  const dispatch = useDispatch()
+  //let networks = getNetworks()
+
+  // Recieve the data to be migrated and save it into the localstorage
+  useEffect(() => {
+    const saveEventData = (event: MigrationMessageEvent) => {
+      const isTrustedOrigin = networks.some((network) => {
+        return network.safeUrl.includes(event.origin)
+      })
+      const isRightOrigin = event.origin !== self.origin && isTrustedOrigin
+      if (event.data.migrate && isRightOrigin) {
+        try {
+          const payload = JSON.parse(event.data.payload)
+          Object.keys(payload).forEach((key) => {
+            const payloadEntry = JSON.parse(payload[key])
+            if (key === 'SAFE__addressBook') {
+              dispatch(addressBookImport(JSON.parse(payloadEntry)))
```

Comment:
What's the behavior of the import action for the same address on different networks?
E.g. on Rinkeby, I have:
`0x453464654654y;Alice Rinkeby;4`
And on Polygon:
`0x453464654654y;Alice Polygon;137`
How will it merge them?

Reply:
I would expect this to work as if you were importing a csv with same addresses but different chains

Reply:
Yes, but how exactly will it merge the example above?

Reply:
If the chainId and address are different it will add a new entry, if not it will replace the name

Reply:
You mean the chainId OR address?

Reply:
Oh sorry. Yeah exactly, if the chainId and address are the same it will replace the name only, if a new entry will be created